legal-api/src/legal_api/models/legal_entity.py

```diff
@@ -786,9 +786,22 @@ def find_by_legal_name(cls, legal_name: str = None):
     @classmethod
     def find_by_identifier(cls, identifier: str = None):
         """Return a Business by the id assigned by the Registrar."""
+        if not identifier or not cls.validate_identifier(
```

Comment:
This sync with the method at common data model https://github.com/bcgov/lear/blob/feature-legal-name/python/common/business-registry-model/src/business_model/models/legal_entity.py#L931